subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/PlainTextBuildVerdictWriter.kt

```diff
@@ -2,29 +2,23 @@ package com.avito.android.build_verdict.internal.writer
 
 import com.avito.android.build_verdict.internal.BuildVerdict
 import com.avito.utils.logging.CILogger
-import org.gradle.api.file.Directory
-import org.gradle.api.provider.Provider
 import java.io.File
 
 internal class PlainTextBuildVerdictWriter(
-    private val buildVerdictDir: Provider<Directory>,
-    private val logger: CILogger
-) : BuildVerdictWriter {
+    buildVerdictDir: Lazy<File>,
+    override val logger: CILogger
+) : BuildVerdictWriter(buildVerdictDir, fileName) {
 
-    override fun write(buildVerdict: BuildVerdict) {
-        val dir = buildVerdictDir.get().asFile.apply { mkdirs() }
-        val file = File(dir, buildVerdictFileName)
-        file.createNewFile()
-        file.writeText(
+    override fun writeTo(buildVerdict: BuildVerdict, destination: File) {
+        destination.writeText(
             when (buildVerdict) {
                 is BuildVerdict.Execution -> buildVerdict.plainText()
                 is BuildVerdict.Configuration -> buildVerdict.plainText()
             }
         )
-        logger.warn("Pretty build verdict at $file")
     }
 
     companion object {
-        const val buildVerdictFileName = "pretty-build-verdict.txt"
+        const val fileName = "pretty-build-verdict.txt"
```

Comment:
it looks like a constructor property for writers, why static?

Reply:
It's used in tests and as a constructor argument